GraphSturcture/src/graphStructure/Graph.java

```diff
@@ -0,0 +1,192 @@
+      unusedId = maxVertexId;
+    }
+    return unusedId;
+  }
+
+  private int firstEdgeIdUnused() {
+    int unusedId;
+    if (unusedEdgeId.size() != 0) {
+      unusedId = unusedEdgeId.poll();
+    } else {
+      maxEdgeId++;
+      unusedId = maxEdgeId;
+    }
+    return unusedId;
+  }
+
+  public void addVertex() {
+    Integer newVertexId = new Integer(firstVertexIdUnused());
+    Vertex newVertex = new Vertex(newVertexId);
+    vertices.put(newVertex.getId(), newVertex);
+  }
+
+  public void deleteVertex(int deVertexId) {
+    if (!vertices.containsKey(deVertexId)) {
+      throw new NullPointerException("The vertex id is not existed");
+    }
+
+    /** *The foreach loop traverse all the edges connected with the deleted vertex,and delete the deleted vertex's Id from all of its
+     adjacent vertices,and delete all edges connectting the deleted vertices and its adjacent vetices(These edges will be removed not only
+     from the HashMap "edges" ,but also be removed from the deleted vertex's adjacent vertices's HashMap"adjEdges")* **/
+    for (Object key : vertices.get(deVertexId).getAdgEdges().keySet()) {
+      Integer adjVerId =
+          edges.get(key).getFrom().getId().intValue() == deVertexId ? edges.get(key).getTo().getId()
+              : edges.get(key).getFrom()
+                  .getId();//to determine wether the adjVertex is the FromVertex of The edge or the ToVertex of the edge.
+      vertices.get(adjVerId).deleteAdjVertex(deVertexId);//
+      vertices.get(adjVerId).deleteAdjEdge(Integer.parseInt(key.toString()));
+      edges.remove(key);
+      unusedEdgeId.add(Integer.parseInt(key.toString()));
+    }
+    vertices.remove(deVertexId);
+    unusedVertexId.add(deVertexId);
+  }
+
+  public void addEdge(Integer fromId, Integer toId) {
+    if (fromId == toId) {
+      System.out.println("Loop is not allowed");
+      return;
+    }
+    if (!vertices.containsKey(fromId)) {
+      System.out.println("The fromVertex is not existed");
+    }
+    if (!vertices.containsKey(toId)) {
+      System.out.println("The toVertex is not existed");
+    }
+    if (!this.vertices.get(fromId).isAdjacent(toId)) {
+      this.vertices.get(fromId).addAdjVertex(vertices.get(toId));
+    }
+    if (!this.vertices.get(toId).isAdjacent(fromId)) {
+      this.vertices.get(toId).addAdjVertex(vertices.get(fromId));
+    }
+    Integer newEdgeId = new Integer(firstEdgeIdUnused());
+    Edge newEdge = new Edge(newEdgeId, vertices.get(fromId), vertices.get(toId));
+    edges.put(newEdgeId, newEdge);
+    vertices.get(fromId).addAdjEdge(newEdge);
+    vertices.get(toId).addAdjEdge(newEdge);
+  }
+
+  private void deleteEdge(Integer deEdgeId) {
+    if (!edges.containsKey(deEdgeId)) {
+      throw new NullPointerException("The edge id is not existed");
+    }
+    Integer fromVertexId = edges.get(deEdgeId).getFrom().getId();
+    Integer toVertexId = edges.get(deEdgeId).getTo().getId();
+    vertices.get(fromVertexId).deleteAdjVertex(toVertexId);
```

Comment:
You only need to maintain the adjacency list in Vertex as Map<Integer, Edge<>>. You can either keep the edge's id as the key or the source node's id.